### PR TITLE
fix/PSD-2690_actionview_temp_error

### PR DESCRIPTION
**Pull request by tahir-khalid**
## Description

Resolved no route found error for notifications controller.

PSD-2690

### CODE REVIEW

File: cosmetics-web/app/controllers/responsible_persons/contact_persons_controller.rb [hunk 1]
```diff
@@ -17,6 +17,10 @@ def create
       # we don't want to switch to new RP
       # See also ResponsiblePerspon::AccountWizardController
       set_current_responsible_person_from_previous
+      if current_responsible_person.nil?
```
Comment (meeravnagesh) [inline cosmetics-web/app/controllers/responsible_persons/contact_persons_controller.rb:20]: the code inside if condition , does not do anything , is that a method

File: cosmetics-web/app/helpers/responsible_persons/notifications_helper.rb [hunk 2]
```diff
@@ -409,6 +409,15 @@ def notification_summary_label_image_link(image, responsible_person, notificatio
     end
   end
 
+  def set_responsible_person
```
Comment (meeravnagesh) [inline cosmetics-web/app/helpers/responsible_persons/notifications_helper.rb:412]: the same of the mthod should be get_responsible_person

File: cosmetics-web/app/helpers/responsible_persons/notifications_helper.rb [hunk 3]
```diff
@@ -409,6 +409,15 @@ def notification_summary_label_image_link(image, responsible_person, notificatio
...
+    @responsible_person =
+      if session[:current_responsible_person] && session[:current_responsible_person][:id] == params[:responsible_person_id]
+        ResponsiblePerson.new(session[:current_responsible_person])
```
Comment (meeravnagesh) [inline cosmetics-web/app/helpers/responsible_persons/notifications_helper.rb:415]: session[:current_r_p] is already a object , is there a reason why we are creating a new object here?

File: cosmetics-web/app/controllers/responsible_persons/contact_persons_controller.rb [hunk 4]
```diff
@@ -47,7 +51,7 @@ def update
 private
 
   def set_responsible_person
-    @responsible_person = ResponsiblePerson.find(params[:responsible_person_id])
+    @responsible_person = helpers.set_responsible_person
```
Comment (meeravnagesh) [inline cosmetics-web/app/controllers/responsible_persons/contact_persons_controller.rb:54]: the name of the helper is deceiving as it says set but its assigned to a variable

File: cosmetics-web/app/controllers/responsible_persons/contact_persons_controller.rb (hunk 1 above)
Comment (tahir-khalid) [reply cosmetics-web/app/controllers/responsible_persons/contact_persons_controller.rb:20]: It does from a certain flow and there are a few ways to hit this code if I am right but you could also be correct will make a task to test this and include or exclude that code thanks.

File: cosmetics-web/app/helpers/responsible_persons/notifications_helper.rb (hunk 2 above)
Comment (alan-at-work) [reply cosmetics-web/app/helpers/responsible_persons/notifications_helper.rb:412]: I would agree keep this in the controllers, not need to move it out really.

File: cosmetics-web/app/helpers/responsible_persons/notifications_helper.rb (hunk 3 above)
Comment (alan-at-work) [reply cosmetics-web/app/helpers/responsible_persons/notifications_helper.rb:415]: if we remove the method this comment is void ;)

**Pull request closed without merging**